server/utils/courseParticipationUtils.test.ts

```diff
@@ -20,6 +20,7 @@ describe('CourseParticipationUtils', () => {
           detail: 'Motivation level over 9000!',
           status: 'complete',
           yearCompleted: 2019,
+          yearStarted: 2018,
```

Comment:
Ooh, this is an interesting issue. A `courseParticipation` shouldn't have both a `yearStarted` and a `yearCompleted`. If the outcome is "complete" it has `yearCompleted` but if it's "incomplete" it has `yearStarted`. We should probably add some logic to the factory for this, or a couple of traits.

To get this intermittent flakiness to pass though, let's make this `null`

Reply:
Ah didn't realise that was the case, cheers.

Reply:
I'm confused about why this is happening. I would've though if you pass in an object with a defined `outcome` it would remove the randomness of that property, but it seems not 🤔

Reply:
Yeah, I'm surprised about that too...I have a feeling it merges the params under the hood.

Reply:
I wonder if we need to specify `undefined` explicitly on every missing property in the participation and participation outcome traits